Dragonflight/DawnOfTheInfinite/BlightOfGalakrond.lua

```diff
@@ -81,24 +101,33 @@ end
 -- Stages
 
 function mod:MalignantTransferal(args)
+	if args.spellId == 415097 then -- Stage 2
+		self:CDBar(407159, 30.4) -- Blight Reclamation
+		self:CDBar(406886, 6.9) -- Corrosive Infusion
+		self:CDBar(407978, 16.8) -- Necrotic Winds
+	else -- 415114, Stage 3
+		self:CDBar(407159, 63.8) -- Blight Reclamation
+		self:CDBar(408141, timers[408141][incineratingBlightbreathCount]) -- Incinerating Blightbreath
+		self:CDBar(406886, 15) -- Corrosive Infusion
+		self:CDBar(408029, 31.2) -- Necrofrost
+	end
+end
+
+function mod:MalignantTransferalApplied(args)
 	if args.spellId == 415097 then -- Stage 2
 		self:SetStage(2)
 		self:Message("stages", "cyan", CL.percent:format(80, CL.stage:format(2)), args.spellId)
 		self:PlaySound("stages", "long")
-		-- TODO boss moves to a specific location before transforming, is it better to
-		-- use StopBar here and trigger these timers off something else?
-		self:CDBar(407159, 9.7) -- Blight Reclamation
-		self:CDBar(406886, 18.1) -- Corrosive Infusion
-		self:CDBar(407978, 27.8) -- Necrotic Winds
+		self:StopBar(406886) -- Corrosive Infusion
+		self:StopBar(407159) -- Blight Reclamation
```

Comment:
Did you mean to reset `corrosiveInfusionCount = 1` here as well?

Reply:
no since I only increment corrosive Infusion in p2 and onwards (there was no need for counter based timers in p1)

if you want it for consistency or w/e then I can add it